src/main/java/org/cactoos/text/CapitalizedText.java

```diff
@@ -53,15 +53,15 @@ public CapitalizedText(final Text text) {
     @Override
     public String asString() throws IOException {
         return new JoinedText(
-            new TransformedIterable<String, Text>(
+            new TransformedIterable<Text, Text>(
                 new SplitText(this.origin),
-                new Func<String, Text>() {
+                new Func<Text, Text>() {
                     @Override
-                    public Text apply(final String input) throws Exception {
+                    public Text apply(final Text input) throws Exception {
```

Comment:
Or 6. Option
Declare the Capitalize Function as a "static" (not static as the word static) Field of the Class

```java
public class CapitalizedText implements Text {
    private final Text origin;
    private final Func<Text, Text> func = input -> new FormattedText(
            "%s%s",
            input.asString().substring(0, 1).toUpperCase(),
            input.asString().substring(1)
        );

    public CapitalizedText(final Text text) {
        this.origin = text;
    }
```

I just don't see how an extra constructor would improve the situation.

But are we in agreement that the Function looks nicer as a lambda statement?